src/Anagram.cpp

```diff
@@ -19,7 +25,8 @@ namespace
 
         if ( word.size() == 0 ) return word;
 
-        // sort will not compare things that only have one letter, but it still needs to be lowercase
+        // sort will not compare things that only have one letter, but it still needs to be
+        // lowercase
         if ( word.size() == 1 ) return std::string { std::tolower ( word [0], loc ) };
```

Comment:
`string::size()` is going to return the length in bytes; so this check won't catch single char multibyte codepoints.  Consider mentioning that in the comment, I don't think it's a blocker now though.

Not sure if it's worth figuring out a shorthand comment to indicate that something might not be utf-8 safe yet rather than tracking them down later when there's more code.

Reply:
So that is why `std::string::length` exists!

Reply:
So that is why `std::string::length` exists!

Reply:
Sadly that's not the case, `length()` and `size()` are interchangeable.  It looks like the "easy" way is to convert to utf-32 and count the number of 4 byte words.  It looks like this is one of the rare cases where java got something right.

Reply:
Yeah, I even tried `if ( word.begin() + 1 == word.end() )` but that doesn't work on Ŕ. So I am definitely going to have to rethink this. However, maybe it should block this review. I will make a ticket for this issue.